src/runtime/components/forms/FormGroup.vue

```diff
@@ -0,0 +1,99 @@
+<template>
+  <div :class="[ui.wrapper]">
+    <label>
+      <div v-if="label" :class="[ui.label.wrapper, ui.size[size]]">
+        <p :class="[ui.label.base, required ? ui.label.required : '']">{{ label }}</p>
+        <span v-if="hint" :class="[ui.hint]">{{ hint }}</span>
+      </div>
+      <p v-if="description" :class="[ui.description, ui.size[size]]">{{ description }}</p>
+      <div :class="[label ? ui.container : '']">
+        <slot v-bind="{ error: errorMessage }" />
+        <p v-if="errorMessage" :class="[ui.error, ui.size[size]]">{{ errorMessage }}</p>
+        <p v-else-if="help" :class="[ui.help, ui.size[size]]">{{ help }}</p>
+      </div>
+    </label>
+  </div>
+</template>
+
+<script lang="ts">
+import { computed, defineComponent, provide, inject } from 'vue'
+import type { PropType } from 'vue'
+import { defu } from 'defu'
+import type { FormError } from '../../types'
+
+// TODO: Remove
+// @ts-expect-error
+import appConfig from '#build/app.config'
+
+// const appConfig = useAppConfig()
+
+export default defineComponent({
+  props: {
+    name: {
+      type: String,
+      default: null
+    },
+    size: {
+      type: String,
+      default: null,
+      validator (value: string) {
+        return Object.keys(appConfig.ui.formGroup.size).includes(value)
+      }
+    },
+    label: {
+      type: String,
+      default: null
+    },
+    description: {
+      type: String,
+      default: null
+    },
+    required: {
+      type: Boolean,
+      default: false
+    },
+    help: {
+      type: String,
+      default: null
+    },
+    error: {
+      type: [String, Boolean],
+      default: null
+    },
+    hint: {
+      type: String,
+      default: null
+    },
+    ui: {
+      type: Object as PropType<Partial<typeof appConfig.ui.formGroup>>,
+      default: () => appConfig.ui.formGroup
+    }
+  },
+  setup (props) {
+    const appConfig = useAppConfig()
+    const ui = computed<Partial<typeof appConfig.ui.formGroup>>(() => defu({}, props.ui, appConfig.ui.formGroup))
+
+    const formErrors = inject<Ref<FormError[]> | null>('form-errors', null)
+    const errorMessage = computed(() => {
```

Comment:
Can we just not call it `error` here as it's provided this way?

Reply:
I renamed it do avoid dupe keys warning in the return statement, we can rename it to error if you prefer